main.js

```diff
@@ -121,3 +137,87 @@ app.ports.newGame.subscribe(async ([nonce, gameName]) => {
+  console.log("Register as user", username);
+  let startResp = await fetch(`http${ssl ? 's' : ''}://${host}/login/register/start`, {
+    method: 'POST',
+    body: JSON.stringify({username}),
+    headers: {
+      'Content-Type': 'application/json'
+    }
+  });
+  let { challenge, playerId } = await startResp.json();
+
+  // This should be a challenge, which we should complete
+  let credential = await navigator.credentials.create({
+    publicKey: {
+      challenge: parseHexString(challenge),
+      rp: { name: "Plank Server" },
+      user: { // TODO: Track information about the user id locally?
+        id: parsePlayerId(playerId),
+        name: username,
+        displayName: username
+      },
+      pubKeyCredParams: [ {type: "public-key", alg: -7} ]
+    }
+  });
+  console.log(credential);
+
+  const response = credential.response;
+
+  // Access authenticator data ArrayBuffer
+  console.log('authenticatorData', response.authenticatorData);
+
+  // Access client JSON
+  console.log('clientJSON', clientDataJSON);
+
+  // Access signature ArrayBuffer
+  console.log('clientJSON', response.signature);
+
+  // Access userHandle ArrayBuffer
+  console.log('clientJSON',  response.userHandle);
+
+  // Now send the server back the credential to complete the process
+  // let completeResp = await fetch(`http${ssl ? 's' : ''}://${host}/login/register/complete`, {
+  //   method: 'POST',
+  //   body: JSON.stringify({username, playerId}),
+  //   headers: {
+  //     'Content-Type': 'application/json'
+  //   }
+  // });
+  
+  // TODO: Complete the registration process
+
+  // sessionStorage.setItem('session', JSON.stringify(json));
+  // app.ports.sessionReceive.send(json);
+});
+
+app.ports.authLogin.subscribe(async (username) => {
+  // TODO: Show error if not supported
+  // let challenge0 = new Uint8Array([139, 66, 181, 87, 7, 203, 181, 87, 7, 203, 181, 87, 7, 203, 181, 87, 7, 203, 181, 87, 7, 203, 181, 87, 7, 203, 181, 87, 7, 203, 181, 87, 7, 203, 181, 87, 7, 203, 181, 87, 7, 203, 181, 87, 7, 203, 203, 181, 87, 7, 203, 203, 181, 87, 7, 203, 203, 181, 87, 7, 203, 203, 181, 87, 7, 203, 203, 181, 87, 7, 203]);
+
+  // let credential0 = await navigator.credentials.create({
+  //   publicKey: {
+  //     challenge: challenge0,
+  //     rp: { id: "localhost", name: "Plank Server" },
+  //     user: {
+  //       id: new Uint8Array([79, 252, 83, 72, 214, 7, 89, 26]),
+  //       name: "jamiedoe",
+  //       displayName: "Jamie Doe"
+  //     },
+  //     pubKeyCredParams: [ {type: "public-key", alg: -7} ]
+  //   }
+  // });
+  // console.log(credential0);
+
+  let challenge1 = new Uint8Array([140, 66, 181, 87, 7, 203, 181, 87, 7, 203, 181, 87, 7, 203, 181, 87, 7, 203, 181, 87, 7, 203, 181, 87, 7, 203, 181, 87, 7, 203, 181, 87, 7, 203, 181, 87, 7, 203, 181, 87, 7, 203, 181, 87, 7, 203, 203, 181, 87, 7, 203, 203, 181, 87, 7, 203, 203, 181, 87, 7, 203, 203, 181, 87, 7, 203, 203, 181, 87, 7, 203]);
```

Comment:
🧐

Reply:
Haha, slightly better up here: https://github.com/hayesgm/plank/pull/4/files#diff-58417e0f781b6656949d37258c8b9052ed266e2eb7a5163cad7b0863e6b2916aR155 but the bigger Q is how do I verify the authenticity of the signature on the server. There are some libraries, but I just kinda want to see if there's like a 20-line raw way to do it versus pulling in who knows how many thousands of lines of code from some lib.